### PR TITLE
Removing/fixing TODO comments.

### PR DESCRIPTION
**Pull request by judy2k**
Removed a few TODO comments, and fixed a few more.

Significant changes were:
* Rename `Endpoint` to `CallEndpoint`
* Throw a `NexmoUnexpectedException` from `RequestSigning.verifyRequestSignature`

### CODE REVIEW

File: src/test/java/com/nexmo/client/voice/endpoints/CreateCallMethodTest.java [hunk 1]
```diff
@@ -87,7 +87,6 @@ public void testParseResponse() throws Exception {
                 new BasicStatusLine(new ProtocolVersion("1.1", 1, 1), 200, "OK")
         );
 
-        // TODO: The following deep inspection may be better implemented under CallEventTest
```
Comment (ChrisGuzman) [inline src/test/java/com/nexmo/client/voice/endpoints/CreateCallMethodTest.java]: Do we want to implement a `CallEventTest`?

File: src/test/java/com/nexmo/client/voice/endpoints/ListCallsMethodTest.java [hunk 2]
```diff
@@ -83,7 +83,6 @@ public void parseResponse() throws Exception {
                 new BasicStatusLine(new ProtocolVersion("1.1", 1, 1), 200, "OK")
         );
 
-        // TODO: The following deep inspection may be better implemented under CallEventTest
```
Comment (ChrisGuzman) [inline src/test/java/com/nexmo/client/voice/endpoints/ListCallsMethodTest.java]: Do we want to implement a `CallEventTest`?

File: src/test/java/com/nexmo/client/voice/endpoints/ReadCallMethodTest.java [hunk 3]
```diff
@@ -65,7 +65,6 @@ public void parseResponse() throws Exception {
                 new BasicStatusLine(new ProtocolVersion("1.1", 1, 1), 200, "OK")
         );
 
-        // TODO: The following deep inspection may be better implemented under CallEventTest
```
Comment (ChrisGuzman) [inline src/test/java/com/nexmo/client/voice/endpoints/ReadCallMethodTest.java]: Do we want to implement a `CallEventTest`?

File: src/test/java/com/nexmo/client/voice/endpoints/CreateCallMethodTest.java (hunk 1 above)
Comment (judy2k) [inline src/test/java/com/nexmo/client/voice/endpoints/CreateCallMethodTest.java]: Maybe at some point in the future. At the moment any obvious problems would be caught by these higher level tests, that also serve the purpose of testing the integration between the objects.

File: src/test/java/com/nexmo/client/voice/endpoints/ListCallsMethodTest.java (hunk 2 above)
Comment (judy2k) [inline src/test/java/com/nexmo/client/voice/endpoints/ListCallsMethodTest.java]: Answer above :)

File: src/test/java/com/nexmo/client/voice/endpoints/ReadCallMethodTest.java (hunk 3 above)
Comment (judy2k) [inline src/test/java/com/nexmo/client/voice/endpoints/ReadCallMethodTest.java]: Answer above :)

**Pull request merged**